org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/TextDocumentService.java

```diff
@@ -520,4 +526,39 @@ default void willSave(WillSaveTextDocumentParams params) {}
 	default CompletableFuture<List<SelectionRange>> selectionRange(SelectionRangeParams params) {
 		throw new UnsupportedOperationException();
 	}
+
+	/**
+	 * The {@code textDocument/semanticTokens/full} request is sent from the client to the server to return
+	 * the semantic tokens for a whole file.
+	 */
+	@JsonRequest(value="textDocument/semanticTokens/full")
```

Comment:
Unless you also set `useSegment=false`, this will end up being `textDocument/textDocument/semanticTokens/full`. This is why I left out the `textDocument/` in my suggestion. Same goes for others below.

Reply:
Now I see, thanks!